pkg/runtime/adoption_reconciler.go

```diff
@@ -224,10 +224,16 @@ func (r *adoptionReconciler) Sync(
 	}, described.RuntimeObject()); err != nil {
 		if apierrors.IsNotFound(err) {
 			// If Adopted AWS resource was not found in k8s, create it.
+
+			// Before creation, Keep the copy of original described object
+			// because after the create call, Status gets set to empty
+			describedCopy := described.DeepCopy()
 			if err := r.kc.Create(ctx, described.RuntimeObject()); err != nil {
```

Comment:
Why not just use the described copy in the create?

Reply:
describedCopy is used only for keeping a copy of Status.

* The create call will modify the `described` object because it's passed by reference
* We use the same modified "described" object for status patching
* But before status patching, we copy the original status from `describedCopy`

Does it answer your question? LMK if you were asking for something else.